File: PDOO/DeepSpaceRuby/lib/WeaponType.rb
#encoding: UTF-8

module Deepspace
	module WeaponType
		class Type
			
			def power
				@power
			end
				
			def initialize(power)
				@power = power
			end
		end
		
		LASER = Type.new(2.0)
		MISSILE = Type.new(3.0)
		PLASMA = Type.new(4.0)
	end
end
		
		
			

File: deepspace/WeaponType.java
/*
 * @author: Jesús García León
 * @file: WeaponType
 */
package deepspace;

/**
 * @brief Representa a los tipos de armas del juego
 */
public enum WeaponType{
    LASER((float)2.0), MISSILE((float)3.0), PLASMA((float)4.0);
    
    private final float power;
    
    WeaponType(float power){
        this.power = power;
    }
    
    float getPower(){
        return power;
    }    
}
